index.js

```diff
@@ -149,6 +149,45 @@ switch (process.platform) {
       }
     })
     break
+
+  case 'linux':
+    Object.assign(fsAdmin, {
+      clearAuthorizationCache () {
+        spawnSync('/bin/pkcheck', ['--revoke-temp'])
+      },
+
+      createWriteStream (filePath) {
+        // Prompt for credentials synchronously to avoid creating multiple simultaneous prompts.
+        if (!fsAdmin.testMode && spawnSync('/usr/bin/pkexec', ['/bin/dd']).status !== 0) {
+          const result = new EventEmitter()
+          result.write = result.end = function () {}
+          process.nextTick(() => result.emit('error', new Error('Failed to obtain credentials')))
+          return result
+        }
+
+        const dd = fsAdmin.testMode
+          ? spawn('/bin/dd', ['of=' + filePath])
+          : spawn('/usr/bin/pkexec', ['/bin/dd', 'of=' + filePath])
```

Comment:
Might also need to handle `dd.on('error')` here and propagate that to `stream`, in case `/usr/bin/pkexec` generates an `ENOENT`, like above.

Reply:
I am not sure we need to worry about this, because the synchronous command invocation would have failed prevented `spawn` from having been called if `pkexec` did not exist. I guess there's a chance `pkexec` gets uninstalled after invoking `spawnSync` but that seems pretty unlikely. What do you think? 💭

Reply:
If you think they're either "both present" or "neither present" to the point that we can rely on that first check, then 👍 to not checking again here.